### ISSUE TITLE
[Bug] Firebase Functions always fail with an INTERNAL exception 

### ISSUE THREAD

**Issue by MileyHollenberg** [firebase/firebase-unity-sdk#662]
### [REQUIRED] Please fill in the following fields:

  * Unity editor version: 2021.3.11f1
  * Firebase Unity SDK version: 10.6.0
  * Source you installed the SDK: .unitypackage
  * Problematic Firebase Component: Functions
  * Other Firebase Components in use: Auth, Crashalytics, RemoteConfig, Analytics
  * Additional SDKs you are using: IronSource, Unity Ads
  * Platform you are using the Unity editor on: Mac
  * Platform you are targeting: iOS
  * Scripting Runtime: IL2CPP
  * Pre-built SDK from the [website](https://firebase.google.com/download/unity) or open-source from this repo: Pre-build from the website

### [REQUIRED] Please describe the issue here:

I'm trying to get a very basic `helloWorld` function to work through the Firebase Functions SDK for Unity. I'm able to call the endpoints myself via any HTTP handler but for some reason when using the official SDK it always gives me an exception without any useful information.

My current function (mostly just for testing purposes) in Unity is as follows:

```
		private async void InitializeFirebase()
		{
			Debug.Log("Setting up Firebase Auth");
			string json = AssetDatabase.LoadAssetAtPath<TextAsset>("Assets/google-services.json").text;
			AppOptions options = AppOptions.LoadFromJsonConfig(json);
			FirebaseApp app = FirebaseApp.Create(options);
			DependencyStatus status = await FirebaseApp.CheckAndFixDependenciesAsync();
			auth = FirebaseAuth.GetAuth(app);
			auth.StateChanged += AuthStateChanged;
			await auth.SignInAnonymouslyAsync();

			try
			{
				FirebaseFunctions functions = FirebaseFunctions.GetInstance(app, "europe-west1");
				HttpsCallableReference func = functions.GetHttpsCallable("helloWorld");
				Task<HttpsCallableResult> task = func.CallAsync().ContinueWithOnMainThread(task =>
				{
					if (task.IsCanceled)
					{
						Debug.LogError("Task was cancelled.");
						return null;
					}

					if (task.IsFaulted)
					{
						Debug.LogError("Task encountered an error."); 
						return null;
					}

					return task.Result;
				});
				await task;
				Debug.Log(task.Result.Data);
			}
			catch (FirebaseException e)
			{
				Debug.LogException(e);
			}
		}
```

I've also tried using the DefaultInstance but it gives the same error. 

The function is in Typescript and is the basic `helloWorld` function that comes with setting up a firebase project via the CLI

```
import * as functions from "firebase-functions";

// Start writing functions
// https://firebase.google.com/docs/functions/typescript

export const helloWorld = functions
	.region("europe-west1")
	.https
	.onRequest((request, response) => {
		functions.logger.info("Hello logs!", { structuredData: true });
		response.send("Hello from Firebase!");
	});
```

The function was deployed via the CLI

```
firebase deploy --only functions
```

Here's a screenshot of the entire exception

<img width="1341" alt="Screenshot 2023-03-29 at 20 33 08" src="https://user-images.githubusercontent.com/799917/228648142-eae22a63-d572-4cec-b70d-492e1da9c954.png">

And I've confirmed in Postman that I can call the function both with a GET and POST method without issues. The calls from the SDK do not even reach any server as not a single node has spun up and my logs have remained empty outside of the Postman calls.

I feel that there's an issue with my `FirebaseApp` setup as the name is always `__FIRAPP_DEFAULT` which feels incorrect. The `google-services.json` file was downloaded from the dashboard and I also have the `GoogleService-Info.plist` in my Assets directory. The Authentication package is working and I am able to get both my userID and token from it but for some reason using the Firebase Functions through the SDK just isn't working.

I'm testing this on macOS atm but I've also tried it on Windows before with no luck. Am I missing something here or is this a bug?

**Comment by paulinon** [comment]
Hi @MileyHollenberg,

Thanks for reporting this issue. Could you provide the error code and message received by the client so we can identify the type of error you're facing? You may modify your code accordingly so that you won't receive any `INTERNAL` messages.

```
if (task.IsFaulted) {
    foreach (var inner in task.Exception.InnerExceptions) {
      if (inner is FunctionsException) {
        var e = (FunctionsException) inner;
        // Function error code, will be INTERNAL if the failure
        // was not handled properly in the function call.
        var code = e.ErrorCode;
        var message = e.ErrorMessage;
      }
    }
  }
```

Could you also provide a [minimal, reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of your implementation so that we can have a baseline for troubleshooting? You may upload this to a GitHub repository and add me as a collaborator. Make sure that it's free from any personally identifiable information.

**Comment by MileyHollenberg** [comment]
Created a repo project, https://github.com/MileyHollenberg/FirebaseTest, obviously I removed the `google-services` files from it for security reasons.

The extra code to read out the exception didn't reveal anything useful either, it just said `FunctionsException: Internal INTERNAL` and nothing else. This is also happening in the test project.

**Comment by MileyHollenberg** [comment]
Oh, one interesting development, with the test project the function is now actualy being called, I can see it happening in the logs. The SDK still gives me the `FunctionsException: Internal INTERNAL` exception data though

**Comment by MileyHollenberg** [comment]
I've found the issue, it was a small and sneaky one.

I was using [onRequest](https://firebase.google.com/docs/reference/functions/firebase-functions.https#httpsonrequest) which apparently is purely for HTTP usage whereas [onCall](https://firebase.google.com/docs/reference/functions/firebase-functions.https#httpsoncall) is meant for usage in the Firebase SDK.

I was using a separate tutorial to create the function than the one for connecting it to Unity. [This](https://firebase.google.com/docs/functions/callable) one instead shows the right way to create the function that's callable from the SDK and that's where I saw that I was using `onRequest` rather than `onCall`. After deleting the function and re-creating it the proper way it started working within my test project :D

Guess RTFM really helps if you read the right manuals 😅
